README.md

```diff
@@ -0,0 +1,63 @@
+# Hcommon-ssh
+
+This library provides features to allow network-connected components to go through SSH tunnels.
+
+## Start using
+
+You can obtain this library from Maven Central:
+
+[![Maven Central TGZ](https://maven-badges.herokuapp.com/maven-central/com.hotels/hcommon-ssh/badge.svg?subject=com.hotels:hcommon-ssh.jar)](https://maven-badges.herokuapp.com/maven-central/com.hotels/hcommon-ssh) [![Build Status](https://travis-ci.org/HotelsDotCom/hcommon-ssh.svg?branch=master)](https://travis-ci.org/HotelsDotCom/hcommon-ssh) [![Coverage Status](https://coveralls.io/repos/github/HotelsDotCom/hcommon-ssh/badge.svg?branch=master)](https://coveralls.io/github/HotelsDotCom/hcommon-ssh?branch=master) ![GitHub license](https://img.shields.io/github/license/HotelsDotCom/hcommon-ssh.svg)
+
+## Overview
+
+Components can be proxied to use a SSH tunnel. A `SshSettings` object provides the basic configuration required to establish the SSH connection and the SSH tunnel.
+
+The library uses Java Proxies to create handle the SSH tunneling logic so components must implement the marker interface `Tunnelable`.
+
+An out-of-the-box factory is provided to handle the component wrapping: `TunnelableFactory`. The factory uses a `MethodChecker` to verify whether the method being invoked must ensure the SSH tunnel is open or whether the method must close the SSH tunnel.
+
+In oder to proxy a component the `wrap` method of this factory class must be invoked passing component to wrap, an instance of `MethodChecker` and the remote host and port to which the component will be connecting.
+
+Example:
+
+    // Mark the client as tunnelable
+    public class MyClient implenents Tunnelable, ... {
+      ...
+    }
+    ...
+    MyClient myClient = new MyClient();
+    ...
+    SshSettings sshSettings = SshSettings
+      .builder()
+      .withRoute("hop1 -> hop2")
+      .withKnownHosts("/home/user/.ssh/known_hosts")
+      .withPrivateKeys("/home/user/.ssh/id_rsa_common")
+      .build();
+    TunnelableFactory factory = new TunnelableFactory(sshSettings);
+    Tunnelable wrapped = factory.wrap(myComponent, MethodCheck.DEFAULT, "my-remote-service", 8080);
```

Comment:
`myComponent` -> `myClient` to match with above?

Reply:
yup, good spot